### ISSUE TITLE
symbolWithContextSync causing Maximum call stack size exceeded

### ISSUE THREAD

**Issue by jlchmura** [mike-lischke/antlr4-c3#117]
Mike,

The call to `getAllSymbolsSync` on line 303 was causing a RangeError in one of my symbol tables.  Shouldn't it have `localOnly` set to true?  Unless I misunderstand this use of this function, if we're looking for a symbol tied to a specific context, why would that symbol be in a non-local table anyway? 

https://github.com/mike-lischke/antlr4-c3/blob/c0530ed7e41911e734a5be75abf5d381589398b5/src/SymbolTable.ts#L303

**Comment by mike-lischke** [comment]
Yes, using only local symbols sounds reasonable. Did it solve your problem with that flag?

But even if it does, with deeply nested symbols this can always crash. Better would be to implement that as iteration (and exchange the duplicated local method code by a single implementation).

**Comment by jlchmura** [comment]
Using local only did solve the problem. Though you're correct, I do need to look into an iteration-based implementation because I do have deeply nested symbols and this is bound to pop up somewhere else.
